pkg/model/workflow.go

```diff
@@ -0,0 +1,91 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/puppetlabs/relay/pkg/config"
+)
+
+type WorkflowIdentifier struct {
+	Name string `json:"name"`
+}
+
+type WorkflowSummary struct {
+	*WorkflowIdentifier
+
+	Description string `json:"description"`
+}
+
+type Workflow struct {
+	*WorkflowSummary
+
+	CreatedAt      *time.Time       `json:"created_at"`
+	UpdatedAt      *time.Time       `json:"updated_at"`
+	LatestRevision *RevisionSummary `json:"latest_revision"`
+	State          *WorkflowState   `json:"state,omitempty"`
+	// TODO: last run
+}
+
+type WorkflowEntity struct {
+	Access   *EntityAccess `json:"access"`
+	Workflow *Workflow     `json:"workflow"`
+}
+
+type WorkflowState struct {
+	Triggers []*WorkflowTriggerState `json:"triggers"`
+}
+
+type WorkflowTriggerState struct {
+	Name     string                      `json:"name"`
+	Revision *RevisionSummary            `json:"revision"`
+	Source   *WorkflowTriggerSourceState `json:"source"`
+}
+
+type PushWorkflowTriggerSourceState struct {
+	Token string `json:"token"`
+}
+
+type ScheduleWorkflowTriggerSourceState struct {
+	ScheduledAt string `json:"scheduled_at"`
+}
+
+type WebhookWorkflowTriggerSourceState struct {
+	Endpoint string `json:"endpoint"`
+}
+
+type WorkflowTriggerSourceState struct {
+	Type     string                              `json:"type"`
+	Status   string                              `json:"status"`
+	Push     *PushWorkflowTriggerSourceState     `json:"push,omitempty"`
+	Schedule *ScheduleWorkflowTriggerSourceState `json:"schedule,omitempty"`
+	Webhook  *WebhookWorkflowTriggerSourceState  `json:"webhook,omitempty"`
+}
```

Comment:
Is any of this used anywhere? I can't seem to find reference to it. Similar in the `revisions.go` file.

Reply:
They are used at present to generate the json outputs if you run the commands with `--out=json`. Eventually I would like to have text formatters that do a little bit more beyond dump json, so establishing these models felt like an important step in that direction.

Reply:
Shouldn't all that be going away, though, with the new triggers work? cc @impl

Reply:
This is the up-to-date stuff

Reply:
Okay, thanks. Sorry for the confusion on my end.